insights/parsers/catalina_log.py

```diff
@@ -0,0 +1,109 @@
+"""
+catalina_log - Log files for Tomcat
+===================================
+
+.. note::
+    The tomcat log files are gotten from the directory specified in the java
+    commands.
+"""
+
+from .. import LogFileOutput, parser
+from insights.specs import catalina_server_log
+from insights.specs import catalina_out
+
+
+@parser(catalina_server_log)
+class CatalinaServerlog(LogFileOutput):
+    """
+    Read the tomcat server log file.
+
+    Note that the standard format of Catalina log lines spreads the information
+    over two lines::
+
+        INFO: Command line argument: -Djava.io.tmpdir=/var/cache/tomcat/temp
+        Nov 28, 2017 2:11:20 PM org.apache.catalina.startup.VersionLoggerListener log
+        INFO: Command line argument: -Djava.util.logging.config.file=/usr/share/tomcat/conf/logging.properties
+        Nov 28, 2017 2:11:20 PM org.apache.catalina.startup.VersionLoggerListener log
+        INFO: Command line argument: -Djava.util.logging.manager=org.apache.juli.ClassLoaderLogManager
+        Nov 28, 2017 2:11:20 PM org.apache.catalina.core.AprLifecycleListener lifecycleEvent
+        INFO: The APR based Apache Tomcat Native library which allows optimal performance in production environments was not found on the java.library.path: /usr/java/packages/lib/amd64:/usr/lib64:/lib64:/lib:/usr/lib
+        Nov 28, 2017 2:11:22 PM org.apache.coyote.AbstractProtocol init
+        INFO: Initializing ProtocolHandler ["http-bio-18080"]
+        Nov 28, 2017 2:11:23 PM org.apache.coyote.AbstractProtocol init
+        SEVERE: Failed to initialize end point associated with ProtocolHandler ["http-bio-18080"]
+
+    However, this parser only recognises single lines.
+    Please refer to its super-class :class:`insights.core.LogFileOutput`.
+
+    When using this parser, consider using a filter or a scan method, e.g.::
+
+        CatalinaServerlog.filters.append('initializing ProtocolHandler')
+        CatalinaServerlog.keep_scan('init_pro', 'initializing ProtocolHandler')
+
+    Examples:
+        >>> CatalinaServerlog.filters.append('Failed to initialize')
+        >>> log = shared[CatalinaServerlog]
```

Comment:
@xiangce since `shared` is no longer used for 3.x is there a better way to demonstrate this in an example?

Reply:
My fault, will correct them. Thank you @bfahr.